api/src/test/java/org/pmiops/workbench/initialcredits/InitialCreditsExpirationServiceTest.java

```diff
@@ -1,26 +1,94 @@
+import static org.mockito.Mockito.doThrow;
+import static org.mockito.Mockito.never;
+import static org.mockito.Mockito.reset;
+import static org.mockito.Mockito.times;
+import static org.mockito.Mockito.verify;
+import static org.mockito.Mockito.when;
 
+import jakarta.mail.MessagingException;
 import java.sql.Timestamp;
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Set;
+import org.junit.jupiter.api.BeforeEach;
 import org.junit.jupiter.api.Test;
 import org.pmiops.workbench.FakeClockConfiguration;
+import org.pmiops.workbench.config.WorkbenchConfig;
+import org.pmiops.workbench.db.dao.UserDao;
+import org.pmiops.workbench.db.dao.UserInitialCreditsExpirationDao;
+import org.pmiops.workbench.db.dao.WorkspaceDao;
 import org.pmiops.workbench.db.model.DbUser;
 import org.pmiops.workbench.db.model.DbUserInitialCreditsExpiration;
+import org.pmiops.workbench.db.model.DbWorkspace;
+import org.pmiops.workbench.mail.MailService;
+import org.pmiops.workbench.model.BillingStatus;
+import org.pmiops.workbench.model.InitialCreditExpirationNotificationStatus;
+import org.pmiops.workbench.test.FakeClock;
 import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.beans.factory.config.ConfigurableBeanFactory;
 import org.springframework.boot.test.autoconfigure.orm.jpa.DataJpaTest;
+import org.springframework.boot.test.context.TestConfiguration;
+import org.springframework.boot.test.mock.mockito.MockBean;
+import org.springframework.boot.test.mock.mockito.SpyBean;
+import org.springframework.context.annotation.Bean;
 import org.springframework.context.annotation.Import;
+import org.springframework.context.annotation.Scope;
 
 @DataJpaTest
 @Import({
-  FakeClockConfiguration.class,
   InitialCreditsExpirationServiceImpl.class,
 })
 public class InitialCreditsExpirationServiceTest {
 
   @Autowired private InitialCreditsExpirationService service;
 
+  @MockBean private MailService mailService;
+
+  @SpyBean private UserDao userDao;
+
+  @SpyBean private WorkspaceDao workspaceDao;
+
+  @MockBean private UserInitialCreditsExpirationDao userInitialCreditsExpirationDao;
+
+  @MockBean private FakeClock fakeClock;
+
+  private static WorkbenchConfig config;
+  private static final Timestamp NOW = Timestamp.from(FakeClockConfiguration.NOW.toInstant());
+  private static final Timestamp NOW_PLUS_ONE_DAY =
+      Timestamp.from(FakeClockConfiguration.NOW.toInstant().plusSeconds(24 * 60 * 60));
+  private static final Timestamp NOW_MINUS_ONE_DAY =
+      Timestamp.from(FakeClockConfiguration.NOW.toInstant().minusSeconds(24 * 60 * 60));
+
+  private DbWorkspace workspace;
+
+  @TestConfiguration
+  static class Configuration {
+    @Bean
+    @Scope(ConfigurableBeanFactory.SCOPE_PROTOTYPE)
+    public WorkbenchConfig config() {
+      return config;
+    }
+  }
+
+  @BeforeEach
+  public void setUp() {
+    reset(userDao, mailService, userInitialCreditsExpirationDao, workspaceDao);
```

Comment:
interesting!  what does this do?

Reply:
This resets a mock so the number of calls and other such interactions are changed back to their initial state.

I was having issues at one point verifying the number of calls some of the mocks had. I was not sure if the tests were bleeding into each other. This turned out not to be necessary. Removed this.